### PR TITLE
Fix method calls on annotations

### PR DESCRIPTION
**Pull request by tdurieux**

### CODE REVIEW

File: src/test/java/spoon/test/annotation/AnnotationTest.java [hunk 1]
```diff
@@ -786,6 +787,14 @@ public void testAbstractAllAnnotationProcessorWithGlobalAnnotation() throws Exce
 		assertEquals(1, methodProcessor.elements.size());
 	}
 
+	@Test
+	public void testAnnotationIntrospection() throws Exception {
+		CtClass<Object> aClass = factory.Class().get(AnnotationIntrospection.class);
+		CtMethod<?> constructor = aClass.getMethod("m");
+		CtStatement statement = constructor.getBody().getStatement(1);
+		assertEquals("annotation.equals(null)", statement.toString());
```
Comment (GerardPaligot) [inline src/test/java/spoon/test/annotation/AnnotationTest.java]: What was the result of `statement.toString()` before your patch?

File: src/test/java/spoon/test/annotation/AnnotationTest.java [hunk 2]
```diff
@@ -786,6 +787,14 @@ public void testAbstractAllAnnotationProcessorWithGlobalAnnotation() throws Exce
 		assertEquals(1, methodProcessor.elements.size());
 	}
 
+	@Test
+	public void testAnnotationIntrospection() throws Exception {
+		CtClass<Object> aClass = factory.Class().get(AnnotationIntrospection.class);
+		CtMethod<?> constructor = aClass.getMethod("m");
```
Comment (GerardPaligot) [inline src/test/java/spoon/test/annotation/AnnotationTest.java]: This isn't a constructor.

File: src/test/java/spoon/test/annotation/AnnotationTest.java (hunk 1 above)
Comment (tdurieux) [inline src/test/java/spoon/test/annotation/AnnotationTest.java]: ```annotation.equals()```

File: src/test/java/spoon/test/annotation/AnnotationTest.java (hunk 2 above)
Comment (tdurieux) [inline src/test/java/spoon/test/annotation/AnnotationTest.java]: Modified

**Pull request merged**